TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1580,3 +1757,82 @@ def parse_layer_as_float(shape, properties, fid, zoom):
             properties['layer'] = layer_float
 
     return shape, properties, fid
+
+
+def drop_features_where(
+        feature_layers, zoom, source_layer=None, start_zoom=0,
+        property_name=None, drop_property=True,
+        geom_types=None):
+    """
+    Drops some features entirely when they have a property
+    named `property_name` and its value is true. Also can
+    drop the property if `drop_property` is truthy. If
+    `geom_types` is present and not None, then only types in
+    that list are considered for dropping.
+
+    This is useful for dropping features which we want to use
+    earlier in the pipeline (e.g: to generate points), but
+    that we don't want to appear in the final output.
+    """
+
+    assert source_layer, 'drop_features_where: missing source layer'
+    assert property_name, 'drop_features_where: missing property name'
+
+    if zoom < start_zoom:
+        return None
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    new_features = []
+    for feature in layer['features']:
+        shape, properties, fid = feature
+
+        matches_geom_type = \
+            geom_types is None or \
+            shape.geom_type in geom_types
+
+        val = None
```

Comment:
Yea, depends on the language. In python, as long as each branch sets the variable to a value, it'll work.

That being said, it's always safer to set it to a default value like you did, especially if the branches get more involved, because if a branch exists that doesn't happen to set the variable, a `NameError` would get raised.